workflows/prognostic_c48_run/prepare_config.py

```diff
@@ -145,7 +139,7 @@ def user_config_from_dict_and_args(config_dict: dict, args) -> UserConfig:
         )
 
     return UserConfig(
-        prephysics=prephysics,
+        prephysics=config_dict.get("prephysics"),
```

Comment:
I think this does exactly what you're suggesting. The dict is converted to a dataclass since that `Union` is defined in the `UserConfig` attributes.

Reply:
I think you'd need an explicit `dacite.from_dict` call to convert it to a dataclass. But it doesn't end up mattering since we call `dataclasses.asdict(user_config)` in line 276 anyway (i.e. for the purposes of serializing, we want the UserConfig as a dict anyway).

Reply:
OK I see the issue you're pointing out: the code as written sent a dict to the `UserConfig` did not check its contents via dacite. But the problem is that dacite can't call `from_dict` with the first argument being a type, it actually has to be a concrete class. That is frustrating. Doing what you suggest requires a hardcoded if statement to decide which type of config in the `Union` is there, which I added. The alternative would be to define a new class which has an attribute that is a `Union` of the two types (dacite can handle this; if you look at their Union example page that's how it's done), but that adds more complexity and confusion I think.

Reply:
I see. Yeah... ideally we would just do a `dacite.from_dict(UserConfig, config_dict)` here—which is what is done at runtime—but the insertion of all the defaults makes this not straightforward. Maybe could change the order, i.e. generate the `UserConfig` dataclass using dacite, and then fill in the custom defaults as necessary.

Anyway, I don't think this should hold up the PR. I'm fine to merge as is... the prepare config script is already a mess of dicts and dataclasses, so this is not making it worse!